### ISSUE TITLE
cvxpy 1.0.3 traceback using CVXOPT

### ISSUE THREAD

**Issue by sunilshah** [cvxgrp/cvxpy#510]
When solving a 3 variable l1-regularized least squares problem, I see python traceback. 

/usr/local/lib/python3.6/site-packages/cvxpy/problems/solvers/cvxopt_intf.py in format_results(self, results_dict, data, cached_data)
    415                     new_dual[:P_rows] = 0
    416                 else:
--> 417                     new_dual[:P_rows] = data["P_leq"].T*z
    418                 new_dual[P_rows:] = new_results[s.INEQ_DUAL][leq_len:]
    419                 new_results[s.INEQ_DUAL] = new_dual

[testcvxopt.html.zip](https://github.com/cvxgrp/cvxpy/files/2065619/testcvxopt.html.zip)

**Comment by SteveDiamond** [comment]
Please don't post zip files. I don't like downloading random zip files.

**Comment by sunilshah** [comment]
I had to zip it because github did not allow me to include .html output from Jupyter notebook.  

How would you like the bug report?   I am trying to be helpful.

**Comment by sunilshah** [comment]
Here it is:
************************
import numpy as np
import cvxpy as cvx
n=3
m=2
k=1
np.random.seed(1)
xTrue=np.random.rand(m,k)
A=np.random.rand(n,m)
b0=A@xTrue
noise=np.random.randn(n,k)
b=b0+noise
xEst=cvx.Variable((m,k))
gamma=cvx.Parameter(nonneg=True)
err=cvx.sum_squares(A*xEst-b)
penalty1=cvx.norm(xEst,1)
obj=cvx.Minimize(err+penalty1)
prob=cvx.Problem(obj)
#  TRACEBACK for l1-regularized penalty
resCVX2=prob.solve(solver='CVXOPT',verbose=True)
--------------------------------  Output  --------
     pcost       dcost       gap    pres   dres   k/t
 0:  0.0000e+00 -1.0000e+00  6e+01  2e+00  1e+00  1e+00
 1:  1.4024e+01  1.6917e+01  1e+02  2e+00  2e+00  5e+00
 2:  6.3266e+00  9.0873e+00  3e+00  3e-01  2e-01  3e+00
 3:  8.6975e+00  9.9456e+00  2e+00  1e-01  9e-02  1e+00
 4:  9.6416e+00  9.6626e+00  6e-02  3e-03  2e-03  2e-02
 5:  9.6542e+00  9.6546e+00  1e-03  7e-05  4e-05  5e-04
 6:  9.6545e+00  9.6546e+00  2e-04  9e-06  5e-06  7e-05
 7:  9.6546e+00  9.6546e+00  8e-06  4e-07  3e-07  2e-06
 8:  9.6546e+00  9.6546e+00  3e-07  1e-08  8e-09  7e-08
Optimal solution found.
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-975f9a0c9d9b> in <module>()
      6 print(obj.value)
      7 #  TRACEBACK for l1-regularized penalty
----> 8 resCVX2=prob.solve(solver='CVXOPT',verbose=True)

/usr/local/lib/python3.6/site-packages/cvxpy/problems/problem.py in solve(self, *args, **kwargs)
    244         else:
    245             solve_func = Problem._solve
--> 246         return solve_func(self, *args, **kwargs)
    247 
    248     @classmethod

/usr/local/lib/python3.6/site-packages/cvxpy/problems/problem.py in _solve(self, solver, ignore_dcp, warm_start, verbose, parallel, **kwargs)
    358         data, inverse_data = self._solving_chain.apply(self)
    359         solution = self._solving_chain.solve_via_data(self, data, warm_start, verbose,
--> 360                                                       kwargs)
    361         self.unpack_results(solution, self._solving_chain, inverse_data)
    362         return self.value

/usr/local/lib/python3.6/site-packages/cvxpy/reductions/solvers/solving_chain.py in solve_via_data(self, problem, data, warm_start, verbose, solver_opts)
    231         """
    232         return self.solver.solve_via_data(data, warm_start, verbose,
--> 233                                           solver_opts, problem._solver_cache)

/usr/local/lib/python3.6/site-packages/cvxpy/reductions/solvers/conic_solvers/cvxopt_conif.py in solve_via_data(self, data, warm_start, verbose, solver_opts, solver_cache)
    106             warm_start,
    107             verbose,
--> 108             solver_opts)

/usr/local/lib/python3.6/site-packages/cvxpy/problems/solvers/cvxopt_intf.py in solve(self, objective, constraints, cached_data, warm_start, verbose, solver_opts)
    192         # Restore original cvxopt solver options.
    193         self._restore_solver_options(old_options)
--> 194         return self.format_results(results_dict, data, cached_data)
    195 
    196     def cpl_solve(self, data, kktsolver):

/usr/local/lib/python3.6/site-packages/cvxpy/problems/solvers/cvxopt_intf.py in format_results(self, results_dict, data, cached_data)
    415                     new_dual[:P_rows] = 0
    416                 else:
--> 417                     new_dual[:P_rows] = data["P_leq"].T*z
    418                 new_dual[P_rows:] = new_results[s.INEQ_DUAL][leq_len:]
    419                 new_results[s.INEQ_DUAL] = new_dual

TypeError: incompatible dimensions
*******************************

By the way, the same problem works with other solvers (ECOS, OSQP).  

On dropping the l-1 penalty term in the objective, CVXOPT succeeds.  

Hope this helps.

**Comment by rileyjmurray** [comment]
I think there are two bugs in ``cvxopt_intf.py``.
1.  On line 409, ``P_rows`` is set to the number of *columns* of a matrix ``data["P_leq"]``.
2.  On line 417, there is a multiplication ``data["P_leq"].T * z``. This disagrees with a comment on line 398, that says we need to multiply duals by ``P_leq`` (no transpose).

I have changed line 409 to correctly get the number of rows, and changing line 417 to remove the transpose. All tests pass (including an additional test I wrote to check dual variables for SOCPs and LPs), and @sunilshah 's example now runs without generating an error.

I will be including these changes in my next pull request.
